src/AutoRest.CSharp/Common/Output/Models/MethodSignatureBase.cs

```diff
@@ -3,15 +3,91 @@
 
 using System;
 using System.Collections.Generic;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Mgmt.AutoRest;
 using AutoRest.CSharp.Output.Models.Shared;
 using AutoRest.CSharp.Utilities;
+using Microsoft.CodeAnalysis;
 
 namespace AutoRest.CSharp.Output.Models
 {
     internal abstract record MethodSignatureBase(string Name, string? Summary, string? Description, MethodSignatureModifiers Modifiers, IReadOnlyList<Parameter> Parameters, IReadOnlyList<CSharpAttribute> Attributes)
     {
         public string? SummaryText => Summary.IsNullOrEmpty() ? Description : Summary;
         public string? DescriptionText => Summary.IsNullOrEmpty() || Description == Summary ? string.Empty : Description;
+
+        public static IEnumerable<MethodSignature> PopulateMethods(INamedTypeSymbol? typeSymbol)
+        {
+            if (typeSymbol is null)
+            {
+                // TODO: handle missing type
+                yield break;
+            }
+            var methods = typeSymbol!.GetMembers().OfType<IMethodSymbol>();
+            foreach (var method in methods)
+            {
+                var typeFactory = new TypeFactory(MgmtContext.Library);
```

Comment:
Need to avoid repetitively creating `TypeFactory`

Reply:
Exposed `TypeFactory` in `MgmtContext` instead.